k8s/dns-ssl-kube-lego/README.md

```diff
@@ -0,0 +1,198 @@
+  before your certificates expire.
+
+```
+    $ EMAIL="..."
+    $ sed -i "s/\[YOUR_EMAIL_ADDRESS_HERE\]/$EMAIL/g" lego/1-configmap.yaml
+```
+
+- Deploy KubeLego:
+
+  ```
+      $ kubectl apply -f lego/
+      namespace "kube-lego" created
+      configmap "kube-lego" created
+      deployment "kube-lego" created
+  ```
+
+2. Configure your Project ID:
+
+```
+    $ PROJECT_ID="<write your project ID here>"
+    $ sed -i "s/\[PROJECT_ID\]/$PROJECT_ID/g" echo/*.yaml
+```
+
+3. Create the service Ingress
+
+```
+    $ kubectl apply -f echo/0-namespace.yaml -f echo/1-ingress.yaml
+    namespace "echo" created
+    ingress "echo-ingress" created
+```
+
+4. Wait for the Ingress controller to assign a public IP address and configure Endpoints DNS accordingly
+
+```
+    $ kubectl get --namespace=echo ingress echo-ingress
+    NAME           HOSTS                                     ADDRESS   PORTS     AGE
+    echo-ingress   echo.endpoints.taton-codelab.cloud.goog             80, 443   33s
+
+    ...
+
+    $ kubectl get --namespace=echo ingress echo-ingress
+    NAME           HOSTS                                     ADDRESS          PORTS     AGE
+    echo-ingress   echo.endpoints.taton-codelab.cloud.goog   130.211.14.157   80, 443   33s
+```
+
+Note: it make take up to a minute for a public IP to be assigned.
+
+```
+    $ INGRESS_IP_ADDRESS="$(kubectl get --namespace=echo ingress echo-ingress \
+          -o jsonpath='{.status.loadBalancer.ingress[0].ip}')"
+    $ sed -i "s/\[INGRESS_IP_ADDRESS\]/$INGRESS_IP_ADDRESS/g" echo/*.yaml
+```
+
+5. Deploy the OpenAPI specification
+
+```
+$ gcloud endpoints services deploy echo/openapi.yaml
+...
+Service Configuration [2017-11-01r0] uploaded for service [echo.endpoints.[PROJECT_ID].cloud.goog]
+```
+
+6. Configure the backend with the service config ID
+
+```
+    $ CONFIG_ID=$(gcloud endpoints configs list \
+        --service=echo.endpoints.$PROJECT_ID.cloud.goog --limit=1 --format="value(id)")
+    $ sed -i "s/\[CONFIG_ID\]/$CONFIG_ID/g" echo/*.yaml
+```
+
+7. Deploy the backend
+
+```
+    $ kubectl apply -f echo/2-service.yaml -f echo/3-deployment.yaml
+    service "echo-service" created
+    deployment "echo-backend" created
+```
+
+8.  Wait for Kube-Lego to provision an SSL/TLS certificates
+
+    It may take in the order of 15 minutes for all the systems and configurations to converge.
+    Eventually, the Ingress controller will be configured with a TLS certificate provisioned by Let's Encrypt.
+
+
+## Verify the service deployment
+
+You can verify the correct operation of the Endpoints service with the following request:
+
+```
+        $ curl --request POST \
+            --header "Content-Type: application/json" \
+            --data '{"message":"hello"}' \
+            https://echo.endpoints.$PROJECT_ID.cloud.goog/echo?key=$API_KEY
+        {
+          "message": "hello"
+        }
+```
+
+Note: you will need to provide a valid API key for your project.
```

Comment:
I'd move this section before the curl, add a link to the credentials page: https://console.cloud.google.com/apis/credentials, and explicitly explain to copypaste the API key and run `export API_KEY=...`

Reply:
Done